Assets/Scripts/skeletonScript.cs

```diff
@@ -49,76 +49,67 @@ protected override void Update()
-                    {
-                        targetEnemy = DetectClosestEnemy();
-                        state = State.Chasing;
-                    }                    
+                    state = State.Moving;
+                    animator.SetBool("isWalking", true);
                 }
 
+                if (enemiesInRange.Count > 0)
+                {
+                    targetEnemy = DetectClosestEnemy();
+                    state = State.Chasing;
+                    animator.SetBool("isWalking", true);
+                }        
+
                 break;
 
             case State.Moving:
-
-                if (enumerator == null)
+                
+                
+                if (agent.remainingDistance <= 0)
                 {
-                    Move();
-
-                    if (agent.remainingDistance <= 0)
-                    {
-                        target = Vector3.zero;
-                        state = State.Idle;
-                    }
-
-                    if (targetEnemy != null)
-                    {
-                        state = State.Chasing;
-                    }
+                    target = Vector3.zero;
+                    state = State.Idle;
+                    animator.SetBool("isWalking", false);
                 }
+                
+                if (targetEnemy != null)
+                {
+                    state = State.Chasing;
+                    animator.SetBool("isWalking", true);
+                }        
 
                 break;
 
             case State.Chasing:
-
-                if (enumerator == null)
+                
+                if (targetEnemy != null)
+                {
+                    target = targetEnemy.transform.position;
+                }
+                
+                Move();
+                
+                float distance = Vector3.Distance(agent.transform.position, target);
+                
+                if (distance < minAttackDistance)
                 {
-                    if (targetEnemy != null)
-                    {
-                        target = targetEnemy.transform.position;
-                    }
-                    
-                    Move();
-
-                    float distance = Vector3.Distance(agent.transform.position, target);
-
-                    if (distance < minAttackDistance)
-                    {
-                        state = State.Grabbing;
-                    }
+                    state = State.Grabbing;
                 }
 
                 break;
 
             case State.Grabbing:
-
-                if (enumerator == null)
+                
+                if (targetEnemy != null)
                 {
-                    if (targetEnemy != null)
-                    {
-                        StartGrabbing();
-                    }
-                    else
-                    {
-                        state = State.Idle;
-                    }
+                    StartGrabbing();
```

Comment:
You can move the startGrabbing method to the chasing state on the distance < minAttackDistance if case and no longer need to check for the enumerater in the startGrabbing method